test/Concurrency/Runtime/async_taskgroup_discarding_noLeaks.swift

```diff
@@ -0,0 +1,119 @@
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library)
```

Comment:
TODO: we need this to run with leak detection

Reply:
I think you'd just add `xcrun leaks -atExit --` to the front of this command, and make sure that this test only runs on macOS.

Reply:
Yeap, needed to add a new lit target but that sounds like a good idea anyway: https://github.com/apple/swift/pull/65613/commits/5ca9af6a0401418f412fb8b5cd3ccca7979e564f

And managed to fix the leak; though one weird edge case about `Error` remains